### ISSUE TITLE
Allow to block for a process that has been Exec'd with ewNoWait

### ISSUE THREAD

**Issue by sschuberth** [jrsoftware/issrc#149]
It would be nice if there was a way to wait for a process to finish that has been `Exec`'d with `ewNoWait`. If calling `CreateProcess` directly, that would be done via `WaitForSingleObject`, but Inno Setup currently does not seem to expose something analogous.

**Comment by TLama** [comment]
If you were **waiting** for a process to finish, it would become blocking again unless you'd be waiting in a thread different from the main one. It would make sense to pass e.g. callback that would execute when the process finishes.

**Comment by sschuberth** [comment]
True. I should have said that in addition to that, I'd like to have a function that checks if a process is still running. That way, one could execute that check in a loop and do some work (e.g. display progress) while the process is still running, without the need to create a separate thread.

**Comment by contre** [comment]
I would also like this kind of behavior. 

My specific situation is that we have to register a type library. This takes a while so it'd be nice if we could launch this registration near the beginning and then wait for it near the end after the rest of the files have been extracted. 

As a workaround, I was leaning towards doing this with two executables. One that starts the long running process and then another that waits for a completion message. This feels like a major hack so I am not looking forward to implementing it.

**Comment by thedarkfalcon** [comment]
I can't believe this feature doesn't exist. I have been using the software for several years and this really grinds my gears that it can't be done. I have a several gigabyte installer that silently installs several other prerequisite applications (database servers etc), and once some of the 3rd party installers have been launched the installer GUI locks up for over 30 minutes while some of the external installers a running, all I can do is leave a "this may take a while, and the installer may appear to freeze" message for the users.

What would be ideal is, launch the 3rd party application via exec with nowait, then wait in a loop checking weather exec has completed. This loop would allow the UI to be updated so the entire application doesn't appear to be frozen.
